apps/rush-lib/src/cli/actions/UpdateAction.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as colors from 'colors';
+import * as os from 'os';
+
+import { CommandLineFlagParameter } from '@microsoft/ts-command-line';
+
+import { BaseRushAction } from './BaseRushAction';
+import { Event } from '../../data/EventHooks';
+import { InstallManager } from '../logic/InstallManager';
+import { PurgeManager } from '../logic/PurgeManager';
+import { RushCommandLineParser } from './RushCommandLineParser';
+import { Stopwatch } from '../../utilities/Stopwatch';
+import { Utilities } from '../../utilities/Utilities';
+
+export class UpdateAction extends BaseRushAction {
+  private _cleanParameter: CommandLineFlagParameter;
+  private _bypassPolicyParameter: CommandLineFlagParameter;
+  private _noLinkParameter: CommandLineFlagParameter;
+  private _fullParameter: CommandLineFlagParameter;
+  private _forceUpdateParameter: CommandLineFlagParameter;
+
+  constructor(parser: RushCommandLineParser) {
+    super({
+      actionName: 'update',
+      summary: '',
+      documentation: '',
+      parser
+    });
+  }
+
+  protected onDefineParameters(): void {
+    this._cleanParameter = this.defineFlagParameter({
+      parameterLongName: '--clean',
+      parameterShortName: '-c',
+      description: ''
+    });
+    this._bypassPolicyParameter = this.defineFlagParameter({
+      parameterLongName: '--bypass-policy',
+      description: 'Overrides "gitPolicy" enforcement (use honorably!)'
+    });
+    this._noLinkParameter = this.defineFlagParameter({
+      parameterLongName: '--no-link',
+      description: ''
+    });
+    this._fullParameter = this.defineFlagParameter({
+      parameterLongName: '--full',
+      description: ''
+    });
+    this._forceUpdateParameter = this.defineFlagParameter({
+      parameterLongName: '--force-update',
+      description: ''
+    });
+  }
+
+  protected run(): Promise<void> {
```

Comment:
>run [](start = 12, length = 3)

This is almost exactly the same as install. There should be a common base class here.

Reply:
Fixed

---
In reply to: [187454349](https://github.com/Microsoft/web-build-tools/pull/646#discussion_r187454349) [](ancestors = 187454349)